crates/rpc/rpc-types/src/eth/trace/parity.rs

```diff
@@ -469,4 +538,153 @@ mod tests {
+        "value": "0x0"
+    },
+    "blockHash": "0xd5ac5043011d4f16dba7841fa760c4659644b78f663b901af4673b679605ed0d",
+    "blockNumber": 18557272,
+    "result": {
+        "address": "0x0292ddd2c4d72bf26602fe5d55de16f50dc535e1",
+        "code": "0x608060405234801561001057600080fd5b50600436106100355760003560e01c8062f55d9d1461003a5780631cff79cd1461004f575b600080fd5b61004d6100483660046101da565b610079565b005b61006261005d3660046101fc565b6100bb565b60405161007092919061027f565b60405180910390f35b6002600054141561009d5760405163caa30f5560e01b815260040160405180910390fd5b600260005573ffffffffffffffffffffffffffffffffffffffff8116ff5b60006060600260005414156100e35760405163caa30f5560e01b815260040160405180910390fd5b600260005573ffffffffffffffffffffffffffffffffffffffff85163b610136576040517f6f7c43f100000000000000000000000000000000000000000000000000000000815260040160405180910390fd5b8473ffffffffffffffffffffffffffffffffffffffff16848460405161015d9291906102de565b6000604051808303816000865af19150503d806000811461019a576040519150601f19603f3d011682016040523d82523d6000602084013e61019f565b606091505b50600160005590969095509350505050565b803573ffffffffffffffffffffffffffffffffffffffff811681146101d557600080fd5b919050565b6000602082840312156101ec57600080fd5b6101f5826101b1565b9392505050565b60008060006040848603121561021157600080fd5b61021a846101b1565b9250602084013567ffffffffffffffff8082111561023757600080fd5b818601915086601f83011261024b57600080fd5b81358181111561025a57600080fd5b87602082850101111561026c57600080fd5b6020830194508093505050509250925092565b821515815260006020604081840152835180604085015260005b818110156102b557858101830151858201606001528201610299565b818111156102c7576000606083870101525b50601f01601f191692909201606001949350505050565b818382376000910190815291905056fea264697066735822122032cb5e746816b7fac95205c068b30da37bd40119a57265be331c162cae74712464736f6c63430008090033",
+        "gasUsed": "0x2cb4a"
+    },
+    "subtraces": 0,
+    "traceAddress": [
+        0,
+        6,
+        0,
+        0
+    ],
+    "transactionHash": "0x54160ddcdbfaf98a43a43c328ebd44aa99faa765e0daa93e61145b06815a4071",
+    "transactionPosition": 102,
+    "type": "create"
+}"#;
+
+        let trace: Value = serde_json::from_str(&reference_data).unwrap();
+        let serialized = serde_json::to_string(&trace).unwrap();
+
+        assert_eq!(serialized, reference_data);
```

Comment:
this check is redundant,

this needs to deserialize into trace object not Value

Reply:
Some issues with this. Is there an easier way to test this than implement deserialize?

```rust
let trace: LocalizedTransactionTrace = serde_json::from_str(&reference_data).unwrap();
let _serialized = serde_json::to_string(&trace).unwrap();
// thread 'eth::trace::parity::tests::test_deserialize_serialize' panicked at crates/rpc/rpc-types/src/eth/trace/parity.rs:685:86:
// called `Result::unwrap()` on an `Err` value: Error("missing field `trace`", line: 25, column: 1)
```

Reply:
possible that I gave you incorrect test data.

will check, but this looks great now!